base/src/main/java/io/spine/type/TypeUrl.java

```diff
@@ -193,12 +193,28 @@ private static String prefixFor(GenericDescriptor descriptor) {
      * Returns a message {@link Class} corresponding to the Protobuf type represented
      * by this type URL.
      *
-     * @return the message class
-     * @throws UnknownTypeException wrapping {@link ClassNotFoundException} if
-     *         there is no corresponding Java class
+     * @return the Java class representing the Protobuf type
+     * @throws UnknownTypeException if there is no corresponding Java class
      */
-    public <T extends Message> Class<T> getJavaClass() throws UnknownTypeException {
-        return KnownTypes.getJavaClass(this);
+    public Class<?> getJavaClass() throws UnknownTypeException {
+        return type().javaClass();
+    }
+
+    /**
+     * Returns a message {@link Class} corresponding to the Protobuf message type represented
+     * by this type URL.
+     *
+     * <p>This is a convenience method. Use it only when sure that the {@link TypeUrl} represents
+     * a message (i.e. not an enum).
+     *
+     * @throws IllegalStateException if the type URL represents an enum
+     */
+    public <T extends Message> Class<T> getMessageClass() throws UnknownTypeException {
```

Comment:
This is an exact duplicate of the same method in `TypeName`. And the one in `TypeName` is not used in `base`. Please review the latter. Maybe it should be removed.